server/artifacts/artifact_server.go

```diff
@@ -344,14 +346,38 @@ func (a *ArtifactServer) httpBadRequestError(w http.ResponseWriter) {
 }
 
 func (a *ArtifactServer) httpFromError(err error, w http.ResponseWriter) {
+	if err == nil {
+		return
+	}
+	statusCode := http.StatusInternalServerError
 	e := &apierr.StatusError{}
-	if errors.As(err, &e) {
+	if errors.As(err, &e) { // check if it's a Kubernetes API error
 		// There is a http error code somewhere in the error stack
-		statusCode := int(e.Status().Code)
-		http.Error(w, http.StatusText(statusCode), statusCode)
+		statusCode = int(e.Status().Code)
 	} else {
-		// Unknown error - return internal error
-		a.serverInternalError(err, w)
+		// check if it's an internal ArgoError
+		switch resolvedError := err.(type) {
+		case argoerrors.ArgoError:
+			switch resolvedError.Code() {
+			case argoerrors.CodeUnauthorized:
```

Comment:
Are ArgoErrors codes and HTTP codes different? can we have same code so it can be send it?

Reply:
So, the argoError code is a string, not an int. I am guessing that Jesse intended that the error be generic and not specific to http. But it may be reasonable to include HTTPCode() into ArgoError, or otherwise to have some generic function which translates ArgoError into an http code, or possibly to move this whole function somewhere more central, since it's not really specific to ArtifactServer.  @jessesuen do you have a preference?

Reply:
@sarabala1979 Maybe no response from Jesse. In that case, we can just decide. It looks like [ArgoError](https://github.com/argoproj/argo-workflows/blob/master/errors/errors.go#L22) is being used in places that are not http related. Do you think it makes sense to add HTTPCode() as a method of the ArgoError interface anyway? Or I can add a function to errors/errors.go which translates an ArgoError into an http code.

Reply:
I would prefer to add HTTPCode in the ArgoError interface. Is ArgoError interface implemented by any other object other than argoerr?

Reply:
don't think so. I added HTTPCode() to ArgoError and argoerr